### PR TITLE
fix: allow migration of pluginsFile using `env` properties

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes UNIFY-1392

### User facing changelog

Allow migrating a project that uses `config.env` by adding a default env property to the config object.

### Details

In some pre-10.X plugins, the config object is modified to add variables to `config.env`. In the migration process, we run the `pluginsFile` function. During this execution, plugins expect the config object contain certain values like "env" since they are provided as default values in 9.X.

EDIT 1: I do not think it is a good idea to copy all the default values from 9 into the migration process. This would mean a lot of code to maintain. I preferred a more pragmatic approach: Check that most common plugins work and fix only the values that are actually used in those. 

EDIT 2: Since I had to diff the output anyway, I finally opted for copying the full defaults from 9.X. I don't expect them to change much and it will be helpful for users. Migration now mirrors as well as possible the way the config is resolved in 9.X.

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by lmiller1990 — DISMISSED** [review]

**Comment by BlueWinds** [comment]
@elevatebart - Can you add details about what this is fixing and why to the PR description?

File: system-tests/projects/migration-e2e-plugins-modify-config/cypress/plugins/index.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module.exports = (on, config) => {
   config.integrationFolder = 'tests/e2e'
   config.testFiles = '**/*.spec.js'
+  // In the plugin @cypress/code-coverage this pattern is used
+  config.env.CYPRESS_RECORD_KEY_DRAGGING = 'true'
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply system-tests/projects/migration-e2e-plugins-modify-config/cypress/plugins/index.js:5]: It does not.
I will indeed add a more detailed description of the changes.

**Review by BlueWinds — APPROVED** [review]

**Comment by lmiller1990** [comment]
@elevatebart looks good - can you get CI green so we can merge? Anything left?

**Comment by elevatebart** [comment]
@lmiller1990 The PR seems ready for review.

I will wait until CI is greeen to send you any slack message and re-request your review.

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
✅

File: packages/data-context/src/sources/migration/legacyOptions.ts [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+}
+
+// NOTE:
+// If you add/remove/change a config value, make sure to update the following
+// - cli/types/index.d.ts (including allowed config options on TestOptions)
+// - cypress.schema.json
+//
+// Add options in alphabetical order for better readability
+
+// TODO - add boolean attribute to indicate read-only / static vs mutable options
```
Comment (ryanthemanuel) [inline packages/data-context/src/sources/migration/legacyOptions.ts:29]: Do we have a followup ticket for this?

**Review by ryanthemanuel — APPROVED** [review]

**Comment by lmiller1990** [comment]
Looks good @elevatebart, once we have an answer on the question above let's merge!!

File: packages/data-context/src/sources/migration/legacyOptions.ts (hunk 2 above)
Comment (elevatebart) [reply packages/data-context/src/sources/migration/legacyOptions.ts:29]: no we don't. this file is a pure copy paste of the 9.X `config/lib/options.ts`. I should clean it up and remove all the not so useful fluff from it.

I could also add where it comes from at the top of the file with a warning to use it only to migrate 9.X projects.

**Pull request merged**